delta-app/src/main/java/io/cdap/delta/store/DraftService.java

```diff
@@ -206,10 +206,15 @@ public TableAssessmentResponse assessTable(DraftId draftId, Configurer configure
     if (target == null) {
       throw new InvalidDraftException("Cannot assess a table without a configured target.");
     }
+    SourceTable selectedTable = deltaConfig.getTables()
+            .stream()
+            .filter(streamTable -> db.equals(streamTable.getDatabase()) && table.equals(streamTable.getTable()))
+            .findAny()
+            .orElse(null);
```

Comment:
I think we need to handle the case where the selectedTable is null, which means the table in the request cannot be found in the draft.

Reply:
Also it would be better add the corresponding test case in the same request to verify your changes :)

Reply:
nit: this and the subsequent null check can be replaced with `.orElseThrow(() -> new IllegalArgumentException(...)))`